refactor(color-picker): Update to use Swatch Group and Swatch

macandcheese opened the pull request:
**Related Issue:** #12733 

## Summary
Updates usage of Color Picker Swatch to instead use Swatch Group and Swatch.

ashetland commented:
Should swatch group be set to selection-mode none? When I click a saved swatch, I'm not really selecting it, but recalling it. Getting some weird behaviors with the current setup. The selection is retained even when I pick a new color and I'm able to toggle my selection with nothing happening when I deselect, which makes sense. Again, not sure "selection" is what I'm actually doing when I click on a saved swatch.

https://github.com/user-attachments/assets/1f6f8f2c-071a-4339-9df0-1fae13660fcb

macandcheese commented:
> Should swatch group be set to selection-mode none? 

Yep - it should be. Updated that and removed some of the other outdated styling. Could you take a look at the new Chromatic?

jcfranco reviewed (approved):
🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥
🟧🟥🟪🟦🟧🟥🟧🟥🟧🟥🟪🟦🟧🟥🟪🟦🟪🟦🟪🟦🟧🟥🟧🟥🟪🟦🟪🟦🟪🟦🟧🟥🟪🟦🟪🟦🟪🟦🟪🟦🟧🟥
🟧🟥🟪🟦🟪🟦🟧🟥🟧🟥🟪🟦🟧🟥🟧🟥🟪🟦🟧🟥🟧🟥🟪🟦🟧🟥🟧🟥🟧🟥🟧🟥🟪🟦🟧🟥🟧🟥🟧🟥🟧🟥
🟧🟥🟪🟦🟧🟥🟪🟦🟧🟥🟪🟦🟧🟥🟧🟥🟪🟦🟧🟥🟧🟥🟪🟦🟧🟥🟧🟥🟧🟥🟧🟥🟪🟦🟪🟦🟪🟦🟧🟥🟧🟥
🟧🟥🟪🟦🟧🟥🟧🟥🟪🟦🟪🟦🟧🟥🟧🟥🟪🟦🟧🟥🟧🟥🟪🟦🟧🟥🟧🟥🟧🟥🟧🟥🟪🟦🟧🟥🟧🟥🟧🟥🟧🟥
🟧🟥🟪🟦🟧🟥🟧🟥🟧🟥🟪🟦🟧🟥🟪🟦🟪🟦🟪🟦🟧🟥🟧🟥🟪🟦🟪🟦🟪🟦🟧🟥🟪🟦🟪🟦🟪🟦🟪🟦🟧🟥
🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥🟧🟥

This is great, @macandcheese! Thanks! 🚀
